src/main/java/ch/njol/skript/structures/StructFunction.java

```diff
@@ -134,4 +169,139 @@ public String toString(@Nullable Event event, boolean debug) {
 		return (local ? "local " : "") + "function";
 	}
 
+	public static class FunctionParser {
+
+		/**
+		 * Parses the signature from the given arguments.
+		 *
+		 * @param script  Script file name (<b>might</b> be used for some checks).
+		 * @param name    The name of the function.
+		 * @param args    The parameters of the function.
+		 * @param returns The return type of the function
+		 * @param local   If the signature of function is local.
+		 * @return Parsed signature or null if something went wrong.
+		 * @see Functions#registerSignature(Signature)
+		 */
+		public static @Nullable Signature<?> parse(String script, String name, String args, @Nullable String returns, boolean local) {
+			SequencedMap<String, Parameter<?>> parameters = parseParameters(args);
+			if (parameters == null)
+				return null;
+
+			// Parse return type if one exists
+			ClassInfo<?> returnClass;
+			Class<?> returnType = null;
+
+            if (returns != null) {
+				returnClass = Classes.getClassInfoFromUserInput(returns);
+				PluralResult result = Utils.isPlural(returns);
+
+				if (returnClass == null)
+					returnClass = Classes.getClassInfoFromUserInput(result.updated());
+
+				if (returnClass == null) {
+					Skript.error("Cannot recognise the type '" + returns + "'");
+					return null;
+				}
+
+				if (result.plural()) {
+					returnType = returnClass.getC().arrayType();
+				} else {
+					returnType = returnClass.getC();
+				}
+
+				return new Signature<>(script, name, parameters, returnType, local);
+			}
+
+			return new Signature<>(script, name, parameters, returnType, local);
+		}
+
+		/**
+		 * Represents the pattern used for the parameter definition in a script function declaration.
+		 * <p>
+		 * The first group specifies the name of the parameter. The name may contain any characters
+		 * but a colon, parenthesis, curly braces, double quotes, or a comma. Then, after a colon,
+		 * the type is specified in the {@code type} group. If a default value is present, this is specified
+		 * with the least amount of tokens as possible.
+		 * </p>
```

Comment:
```suggestion
		 * </p>
		 * name : type = default
		 * [^:(){}\",]+? : [a-zA-Z ]+? = .+
```
including something like this might be helpful, idk